app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt

```diff
@@ -18,27 +16,12 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val moshi = Moshi.Builder()
-            .add(KotlinJsonAdapterFactory())
-            .build()
-
-        val retrofit = Retrofit.Builder()
-            .baseUrl("https://restcountries.com/v3.1/")
-            .addConverterFactory(MoshiConverterFactory.create(moshi))
-            .build()
-
-        val service: CountryService = retrofit.create(CountryService::class.java)
-
-setContent {
-    MyApplicationTheme {
-        CountryInfoScreen(
-            viewModel = viewModel(
-                factory = CountryInfoViewModel.CountryInfoViewModelFactory(
-                    repository = CountryRepositoryImpl(service),
-                ),
-            ),
-        )
-    }
-}
+        setContent {
+            MyApplicationTheme {
+                CountryInfoNavHost(
+                    repository = CountryRepositoryImpl((application as CountryApp).getCountryService())
```

Comment:
note, while this works it's likely you may want to move the entire repository instantiation into the application class you created as well. That way the repository will remain a singleton. If it's left here, the repository will get re-created upon config changes.